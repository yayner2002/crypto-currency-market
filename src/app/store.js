import { configureStore } from "@reduxjs/toolkit";
import coinsReducer from "../redux/crypto/Cryptos";

export const store = configureStore({
  reducer: {
    coins: coinsReducer,
  },
});
