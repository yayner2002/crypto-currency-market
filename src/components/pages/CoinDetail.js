import React from 'react';
import { useParams } from 'react-router-dom';

const CoinDetail = () => {
  const { id } = useParams();
  return (
    <div>
      CoinDetail -
      {' '}
      {id}
    </div>
  );
};

export default CoinDetail;
