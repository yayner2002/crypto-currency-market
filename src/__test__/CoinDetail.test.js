/* eslint-disable no-unused-vars */
import { render } from '@testing-library/react';
import { toBeInTheDocument } from '@testing-library/jest-dom';
import { Provider } from 'react-redux';
import { BrowserRouter as Router } from 'react-router-dom';
import renderer from 'react-test-renderer';
import CoinDetail from '../components/pages/CoinDetail';

import store from '../app/store';

const MockDetail = () => {
  <Provider store={store}>
    <Router>
      <CoinDetail />
    </Router>
  </Provider>;
};

describe('test the detail component', () => {
  it('should render detail component correctlly', () => {
    render(<MockDetail />);
    const tree = renderer.create(<MockDetail />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
