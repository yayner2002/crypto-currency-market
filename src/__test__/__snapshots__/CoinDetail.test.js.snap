// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test the detail component should render detail component correctlly 1`] = `null`;
